src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTarget.java

```diff
@@ -1046,6 +1046,13 @@ public void forceLineHighlighting()
       docDisplay_.setHighlightSelectedLine(true);
    }
    
+   @Override
+   public void setSourceOnSave(boolean sourceOnSave)
+   {
+      view_.getSourceOnSave().setValue(sourceOnSave);
```

Comment:
I think the reason this doesn't sync over to the model is that `setValue` doesn't fire events by default. Does it work with `.setValue(sourceOnSave, true)`?

Reply:
Yes, I just added that and now it works correctly without poking the model!